### ISSUE TITLE
Some special chars in latex labels are not rendered correctly

### ISSUE THREAD

**Issue by dasnessie** [bokeh/bokeh#14732]
### Software versions

Python version        :  3.11.2 (main, Apr 28 2025, 14:11:48) [GCC 12.2.0]
IPython version       :  (not installed)
Tornado version       :  6.5.2
NumPy version         :  2.3.5
Bokeh version         :  3.8.1
BokehJS static path   :  /home/dasnessie/Projects/bokeh-debug/venv/lib/python3.11/site-packages/bokeh/server/static
node.js version       :  v18.20.4
ERROR: [Errno 2] No such file or directory: PosixPath('npm')

### Expected behavior

See also [this discourse thread](https://discourse.bokeh.org/t/special-characters-in-latex-labels/12674)

I’m working on an application that works with different SI units. For that application, I need to display units as bokeh labels in plots. However, I’m having issues with some of the symbols I need - I would expect them to be rendered as they would be if I pasted the latex code into a minimal latex document.

### Observed behavior

I’m having issues with some of the symbols I need, and I’m not sure if the problem is in my usage of the library, or if it’s a bug with bokeh or MathJax.

In particular, I need a non-italic greek lowercase mu (μ) and a uppercase A with an overring (Å), but have been unable to get those. They are rendered almost correctly, but appear very small and on top of the other letters.

### Example code

```Python
from bokeh.plotting import figure, curdoc
from bokeh.models import Label

p = figure(
    width=550,
    height=250,
    x_range=(-50, 500),
    y_range=(-125, 125),
)

label = Label(
    x=0,
    y=0,
    text=(
        r"$$\upmu\mathrm{N}\mathrm{m}$$ "
        r"$$\mbox{µ}\mathrm{N}\mathrm{m}$$ "
        r"$$\text{µ}\mathrm{N}\mathrm{m}$$ "
        r"$$\mbox{µ}^\circ\mathrm{C}\Omega\mbox{Å}\%'''^\circ{\hbar}E_\mathrm{h}$$ "  # these are all the special characters I need. Most of them do work like this.
    ),
)

p.add_layout(label)

curdoc().add_root(p)
```

### Screenshots

<img width="1000" height="454" alt="Image" src="https://github.com/user-attachments/assets/84876740-242e-4c54-9930-9d9a2c9c4501" />

**Comment by bryevdv** [comment]
As noted in the Discourse thread, this content renders correctly on the MathJax demo page, which presumably used MathJax 4.0. We are using 3.2.2 which does seem to have issues with this content in a pure HTML MRE (though not quite the same issues). In any case, seems like we should try updating to 4.0 as a first step. 

https://docs.mathjax.org/en/stable/upgrading/v3.html

Edit: Nominally marking as `task` since I don't think this is a bug on our end.
